src/utility/ota/OTA-portenta-h7.cpp

```diff
@@ -61,7 +64,13 @@ int portenta_h7_onOTARequest(char const * ota_url)
   watchdog_reset();
 
   /* Download the OTA file from the web storage location. */
-  int const ota_portenta_qspi_download_ret_code = ota_portenta_qspi.download(ota_url, true /* is_https */);
+  MbedSocketClass * download_socket = static_cast<MbedSocketClass*>(&WiFi);
+#if defined (ARDUINO_PORTENTA_H7_M7)
```

Comment:
Not sure if this ifdef is really needed? After all, should not this file be only compiled for Portenta H7?

Reply:
Yes, it is. The file is compiled also for NICLA VISION that does not have Ethernet. I think the best thing to do here is to rename the file into `OTA-stm32-h7.cpp` to avoid confusion, but then we need also to rename `Arduino_Portenta_OTA` library; or make a new library called `Arduino_STM32_H7_OTA` and deprecate `Arduino_Portenta_OTA`.